text/0000-build-std/4-build-std-always.md

```diff
@@ -0,0 +1,1260 @@
+`compiler_builtins/mem` trivially overridable in every case
+([?][rationale-no-weak-linkage]).
+
+The `mem` feature of `compiler_builtins` will be inverted to a new feature named
+`external-mem` ([?][rationale-no-mem]). This will not be a default feature, so
+`compiler_builtins` will provide mem symbols unless the `external-mem` is
+provided.
+
+`std`, which provides memory symbols via `libc`, will depend on the
+`external-mem` feature. Most `no_std` users will use the `compiler_builtins`
+implementation of these symbols and will work by default when they do not depend
+on `std`.
+
+Those users providing their own mem symbols can override on weak linkage of the
+`compiler_builtins` symbols. or use a nightly toolchain to enable the
+`external-mem` feature of an explicit dependency on the standard library (per
+[*Standard library dependencies*][deps]).
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why not rely on weak linkage for `compiler-builtins/mem` symbols?*][rationale-no-weak-linkage]
+- [*Why invert the `mem` feature?*][rationale-no-mem]
+
+## `profiler-builtins`
+[profiler-builtins]: #profiler-builtins
+
+`profiler-builtins` will not be built by build-std, thus preventing
+profile-guided optimisation with a locally-built standard library.
+`profiler-builtins` has native dependencies which may fail compilation of the
+standard library if missing were `profiler-builtins` to be built by default as
+part of the standard library build.
+
+*See the following sections for future possibilities:*
+
+- [*Allow building `profiler-builtins`*][future-profiler-builtins]
+
+## Caching
+[caching]: #caching
+
+Standard library artifacts built by build-std will be reused equivalently to
+today's crates/dependencies that are built within a shared target directory. By
+default, this limits sharing to a single workspace ([?][rationale-caching]).
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why not globally cache builds of the standard library?*][rationale-caching]
+
+## Generated documentation
+[generated-documentation]: #generated-documentation
+
+When running `cargo doc` for a project to generate documentation and rebuilding
+the standard library, the generated documentation for the user's crates will
+link to the locally generated documentation for the `core`, `alloc` and `std`
+crates, rather than the upstream hosted generation as is typical for non-locally
+built standard libraries.
+
+*See the following sections for rationale/alternatives:*
+
+- [*Why not link to hosted standard library documentation in generated docs?*][rationale-generated-docs]
+
+## Cargo subcommands
+[cargo-subcommands]: #cargo-subcommands
+
+As opaque dependencies, any Cargo command which accepts a package spec with `-p`
+will only additionally recognise `core`, `alloc` and `std` and none of their
+dependencies. Many of Cargo's subcommands will need modification to support
+build-std:
```

Comment:
Should `-p` support be left to "Explicit dependencies"?  As is, this phase doesn't require any special integration of build-std into how we treat dependencies until we get to this part.  Not saying it can't be done at this phase but feels like it should be left to the implementation while required for "Explicit dependencies".